encoding/codec.go

```diff
@@ -0,0 +1,39 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// Package encoding provides codecs for different []byte encoders and decoders.
+package encoding
+
+import "github.com/elastic/apm-data/model"
+
+// Codec for a specific encoding.
+type Codec interface {
+	Encoder
+	Decoder
+}
+
+// Encoder encodes a model.APMEvent to a []byte
+type Encoder interface {
+	// Encode accepts a model.APMEvent and returns the encoded representation.
+	Encode(model.APMEvent) ([]byte, error)
+}
+
+// Decoder decodes a []byte into a model.APMEvent
+type Decoder interface {
+	// Decode decodes an encoded model.APM Event into its struct form.
+	Decode([]byte, *model.APMEvent) error
+}
```

Comment:
When will we need the Codec interface? Do the Encoder and Decoder interfaces need to be in a common package, or could they be defined in the packages where they are used?

Reply:
I was a bit on the fence on this one, I think we can remove it.